python/kfserving/kfserving/server.py

```diff
@@ -65,6 +82,43 @@ def register_model(self, model):
         self.registered_models[model.name] = model
 
 
+class ModelPredictHandler(tornado.web.RequestHandler):
+    def initialize(self, protocol_handler: RequestHandler, models):
+        self.models = models
+        self.protocol_handler = protocol_handler
+
+    def post(self, name):
+        # TODO Add metrics
+        if name not in self.models:
+            raise tornado.web.HTTPError(
+                status_code=HTTPStatus.NOT_FOUND,
+                reason="Model with name %s does not exist." % name
+            )
+
+        model = self.models[name]
+        if not model.ready:
+            model.load()
+
+        try:
+            body = json.loads(self.request.body)
+        except json.decoder.JSONDecodeError as e:
+            raise tornado.web.HTTPError(
+                status_code=HTTPStatus.BAD_REQUEST,
+                reason="Unrecognized request format: %s" % e
+            )
+
+        # Protocol specific validation and extraction of data
+        self.protocol_handler.validate_request(body)
+        extracted_inputs = self.protocol_handler.extract_inputs(body)
+        # model specific processing
+        inputs = model.preprocess(extracted_inputs)
```

Comment:
For posterity:

Can we free up preprocess/postprocess for their graph concepts?

Reply:
done